lib/matplotlib/tests/test_axes.py

```diff
@@ -2392,12 +2392,17 @@ def test_scatter_color(self):
                                     {'vmax': 0}
                                 ])
     def test_scatter_color_warning(self, kwargs):
+        warn_match = "'c' is empty or not given, "
         # Warn for cases where 'cmap', 'norm', 'vmin', 'vmax'
         # kwargs are being overridden
-        with pytest.warns():
+        with pytest.warns(Warning, match=warn_match):
             plt.scatter([], [], **kwargs)
+            plt.scatter([1,2], [3,4], c=[], **kwargs)
+        # Do not warn for cases where 'c' matches 'x' and 'y'
         with warnings.catch_warnings():
+            warnings.simplefilter('error')
```

Comment:
We fail the tests on any warning, so just running the code will be enough to test that it does not warn.